Db architecture

robot269006 opened the pull request:
# WHAT
1.Conversion of README, from .rdoc to .md
2.Table plannings in detail

# WHY
1.Enable the use of markdown language in README
2.Detailed pre-plannings of the db tables

robot269006 committed:
Convert README

Conversion from .rdoc to .md

robot269006 committed:
DB-planning

Details of DB planning

robot269006 committed:
Merge branch 'master' into DB-architecture

robot269006 committed:
Markdown revision

robot269006 commented:
Markdownを修正しました

robot269006 committed:
Markdown Revision

robot269006 committed:
Markdown Revision

robot269006 commented:
何回か細かく修正が分かれてしまいましたが、最後のファイルをご確認いただけると幸いです

kenchan0503 commented:
中間テーブルはthroughオプションが必要ですね。

kenchan0503 commented:
imageをテーブル単位で分割されておられますが、imageはmessageにのみ付随し、特に大きなデータ構造もないのでカラムとして組み込んで構いません。

robot269006 commented:
イメージテーブルについて承知しました。あくまでコメントの一つというような位置付けですね。
:throughオプションについても承知しました。少し調べて修正します。

robot269006 committed:
DB architecture revision

1.Revised image table, now integrated into comments table
2.Revised mid-table. “:through” option is added

robot269006 commented:
イメージテーブルと:throughオプションの修正をしました。ご確認よろしくお願いいたします。

miyagit commented on README.md:
```diff
@@ -0,0 +1,123 @@
+  ## 1. User
+   | Column        | Type        |
+   |:--|:--|
+   | id            |     integer |
```
id created_at, updated_atはデフォルトで作られるので、書く必要はありません。
他のテーブルに関しても同様によろしくお願いします。

robot269006 replied:
承知しました、こちらは削除します

miyagit commented on README.md:
```diff
@@ -0,0 +1,123 @@
+   | user_id       |     integer |
+   | group_id      |     integer |
+
+  ## 4. Comments
```
Commentsテーブルはどのような用途で必要なのでしょうか？

robot269006 replied:
messageテーブルがスレッドのようなものになり、それに対してコメントを付けていくイメージでした（メッセージというスレッドのようなentityがあり、それに対するコメントというentityがある）。そうではなく、グループがあり、グループに対してコメントを書いていくものなのであれば削除します。

miyagit commented on README.md:
```diff
@@ -0,0 +1,123 @@
+   | body          |        text |
+   | created_at    |      string |
+   | updated_at    |      string |
+   | user_id       |     integer |
```
外部キーには外部キー制約をオプションに付けて下さい。
他のカラムにも同様の設定をお願いします。

robot269006 replied:
外部キーの制約は書く必要があるのでしょうか？最初から前提として設定するものでここで改めて書く必要はないと思ってましたが。。

miyagit commented on README.md:
```diff
@@ -0,0 +1,123 @@
+   | user_id       |     integer |
+   | message_id    |     integer |
+
+  ## 2.5. User-group
```
中間テーブルの命名規則について調べて下さい。
調べましたら、中間テーブルの命名を変更して頂きたく思います。

robot269006 replied:
承知しました、調べて修正します！

miyagit reviewed (commented):
自分で指定せずに作られるカラムに関しては
基本的に書かないようにして頂きたく思います。

robot269006 commented:
自分で指定しないカラム＝idカラムやcreated_atのようなのことでしょうか？idカラムも削除しておきます

robot269006 committed:
Minor revision on DB plan

1. Delete default columns (id, created_at, updated_at) on all tables
2. Changed mid-table namings (User-Group => User_Group)

robot269006 committed:
haml install

narukami894 commented on README.md:
```diff
@@ -0,0 +1,110 @@
+    ### has_many :none
+
+  # Tables
+  ## 1. User
```
user機能に関しては、deviseを使う旨を記述しましょう！

robot269006 replied:
承知しました、ご指摘ありがとうございます

narukami894 commented on README.md:
```diff
@@ -0,0 +1,110 @@
+  ## 4. Comments(chat/image itself)
+    ### belongs_to :user, :message
+    ### has_many :none
+  ## 2.5 User_Group(Mid-table)
```
中間テーブルの命名は、アルファベット順です！

robot269006 replied:
ありがとうございます、修正しました！

narukami894 commented on README.md:
```diff
@@ -0,0 +1,110 @@
+  ## 3. Messages(chat/thread as whole)
+    ### belongs_to :user, :group
+    ### has_many :comments
+  ## 4. Comments(chat/image itself)
```
commentsとmessagesとは何が違うのでしょうか？

robot269006 replied:
昨日もコメントしましたが、必要ないようであればcommentsの方を削除します！

narukami894 commented on README.md:
```diff
@@ -0,0 +1,110 @@
+  ## 2. Group
+    ### belongs_to :none
+    ### has_many :messages
+    ### has_many :users, through: user_group
```
has_many 中間テーブルも記述しましょう！

robot269006 replied:
修正しました！

narukami894 commented on README.md:
```diff
@@ -0,0 +1,110 @@
+  ## 2. Group
+   | Column        | Type        |
+   |:--|:--|
+   | groupname     |        text |
```
こちらgroupsテーブルにあるという時点でgroupのnameであるのは自明のためnameだけで大丈夫です。

robot269006 replied:
修正しました

narukami894 commented on README.md:
```diff
@@ -0,0 +1,110 @@
+   |:--|:--|
+   | groupname     |        text |
+   | body          |        text |
+   | password      |        text |
```
password?

robot269006 replied:
修正しました

narukami894 commented on README.md:
```diff
@@ -0,0 +1,110 @@
+   | groupname     |        text |
+   | body          |        text |
+   | password      |        text |
+   | user-group id |     integer |
```
このカラムは何のためにあるのでしょうか？

robot269006 replied:
どのカラムのことでしょうか？中間テーブルIDのことでしょうか？

narukami894 commented on README.md:
```diff
@@ -0,0 +1,110 @@
+   |:--|:--|
+   | user_id       |     integer |
+   | group_id      |     integer |
+
```
not null制約や、外部キー制約についても記述しましょう！

robot269006 replied:
修正しました

robot269006 committed:
DB revised

1. Mid-table name from “User_group” to “Group_user”
2. Deleted “comments” table as it’s unnecessary
3. Set “has many :user_group” to user and group table
4. Deleted “password” column from group table
5. Add notions of “not null” and “foreign_key” settings

robot269006 commented:
修正させていただきました。ご確認をお願いいたします。
